danny-avila/LibreChat
# Enhanced Placeholder Configuration in LibreChat Prompt Library

## Summary

This pull request introduces enhanced placeholder configuration options for the LibreChat Prompt Library. The changes allow for multiline input fields, dropdown lists with custom input options, and the ability to change the order of placeholder fields on the prompt screen. This feature aims to improve the flexibility and usability of prompts, enabling users to create more sophisticated and tailored prompt templates.
## Change Type

- [ X ] New feature (non-breaking change which adds functionality)
- [ X ] This change requires a documentation update

## Testing
To test this feature:

1. Create a new prompt in the Prompt Library with various placeholder types:
   - Simple text input: `{{name}}`
   - Multiline text input: `{{description:multiline}}`
   - Dropdown with options: `{{tone:formal|casual|business casual}}`

2. Save the prompt and verify that the placeholder configuration is stored correctly.

3. Use the prompt and check if the input form displays the correct field types:
   - Single-line input for `{{name}}`
   - Multiline input for `{{description}}`
   - Dropdown (with custom input option) for `{{tone}}`

4. Verify that the order of placeholders in the input form matches the order in the prompt.

5. Test edge cases:
   - Very long dropdown option lists
   - Nested placeholders (if supported)
   - Special characters in placeholder names and options

### **Test Configuration**:
- Tested on the latest main branch of LibreChat
- Tested with various browsers (Chrome, Firefox)

## Checklist
